src/building/building.h

```diff
@@ -1,8 +1,11 @@
 #ifndef BUILDING_BUILDING_H
 #define BUILDING_BUILDING_H
 
+// TODO why?
+#include <building/depot.h>
```

Comment:
I couldn't figure out why.
If I do 
```
#include "building/depot.h"
```
VS would give me some weird compile errors. https://gist.github.com/RuoanJi/e2cd27397bdaa00ccad0cac650e403ab
However, if I use command line `mingw32-make.exe` to build, it works.

I wonder if it has anything to do with
1. the way I define structs
2. possible loops in header file dependency?

I guess, as last resort, I can always move all the definitions to `building/building.h` and discard `building/depot.h`.

Reply:
I didn't look at the code, but the fact that the main building header includes the depot header doesn't sound right structurally.

The code should be added in a way that any file that uses depots should include it directly, and `building.h` should have depot-free code.

Without looking directly at the code I can't tell you how you can change this, but now you know, I'll ask you to change this 😉